Fix fpp-to-json native build problem

> @jwest115 - I've found several instances where valid fpp files result in java exceptions when using the latest devel commit (402c0317b). For example, https://github.com/fprime-community/fprime-vorago/blob/main/Va416x0/Svc/ExceptionHandler/ExceptionHandler.fpp generates the below exception. 
> 
> Is this issue related to that failure? If not, I'll open a separate issue to track the bug but I wanted to avoid opening a duplicate if this was already being tracked. 
> ```
> # /usr/bin/python3   $PWDlib/fprime/cmake/target/sub-build/../tools/arguments-from-file.py   $PWDbuild-fprime-automatic-va416x0-baremetal/sub-build-info-cache/Va416x0/Svc/ExceptionHandler/fpp-cache/stdout.txt   fpp-to-json   $PWDlib/fprime-vorago/Va416x0/Svc/ExceptionHandler/ExceptionHandler.fpp   
> Exception in thread "main" java.lang.ExceptionInInitializerError
>         at fpp.compiler.codegen.AnalysisJsonEncoder$$anon$1870.elemEncoders(AnalysisJsonEncoder.scala:299)
>         at io.circe.DerivedEncoder.encodeWith(Derivation.scala:237)
>         at io.circe.DerivedEncoder.encodeWith$(Derivation.scala:233)
>         at fpp.compiler.codegen.AnalysisJsonEncoder$$anon$1870.encodeWith(AnalysisJsonEncoder.scala:45)
>         at io.circe.DerivedEncoder$$anon$2.next(Derivation.scala:246)
>         at io.circe.DerivedEncoder$$anon$2.next(Derivation.scala:245)
>         at io.circe.JsonObject$.fromIterable(JsonObject.scala:233)
>         at fpp.compiler.codegen.AnalysisJsonEncoder$$anon$1870.encodeObject(AnalysisJsonEncoder.scala:299)
>         at fpp.compiler.codegen.AnalysisJsonEncoder$$anon$1870.encodeObject(AnalysisJsonEncoder.scala:299)
>         at io.circe.Encoder$AsObject.apply(Encoder.scala:832)
>         at io.circe.Encoder$AsObject.apply$(Encoder.scala:831)
>         at fpp.compiler.codegen.AnalysisJsonEncoder$$anon$1870.apply(AnalysisJsonEncoder.scala:45)
>         at io.circe.DerivedEncoder.encodeWith(Derivation.scala:237)
>         at io.circe.DerivedEncoder.encodeWith$(Derivation.scala:233)
>         at fpp.compiler.codegen.AnalysisJsonEncoder$$anon$1869.encodeWith(AnalysisJsonEncoder.scala:45)
>         at fpp.compiler.codegen.AnalysisJsonEncoder$$anon$1869.encodeObject(AnalysisJsonEncoder.scala:299)
>         at fpp.compiler.codegen.AnalysisJsonEncoder$$anon$1869.encodeObject(AnalysisJsonEncoder.scala:299)
>         at io.circe.Encoder$AsObject.apply(Encoder.scala:832)
>         at io.circe.Encoder$AsObject.apply$(Encoder.scala:831)
>         at fpp.compiler.codegen.AnalysisJsonEncoder$$anon$1869.apply(AnalysisJsonEncoder.scala:45)
>         at io.circe.syntax.package$EncoderOps$.asJson$extension(package.scala:8)
>         at fpp.compiler.codegen.AnalysisJsonEncoder$.f2$30(AnalysisJsonEncoder.scala:299)
>         at fpp.compiler.codegen.AnalysisJsonEncoder$.$init$$$anonfun$34$$anonfun$2(AnalysisJsonEncoder.scala:300)
>         at fpp.compiler.codegen.AnalysisJsonEncoder$.mapAsJsonMap$$anonfun$1(AnalysisJsonEncoder.scala:105)
>         at scala.collection.StrictOptimizedMapOps.map(StrictOptimizedMapOps.scala:28)
>         at scala.collection.StrictOptimizedMapOps.map$(StrictOptimizedMapOps.scala:27)
>         at scala.collection.immutable.HashMap.map(HashMap.scala:39)
>         at fpp.compiler.codegen.AnalysisJsonEncoder$.mapAsJsonMap(AnalysisJsonEncoder.scala:105)
>         at fpp.compiler.codegen.AnalysisJsonEncoder$.$init$$$anonfun$34(AnalysisJsonEncoder.scala:300)
>         at io.circe.Encoder$$anon$20.apply(Encoder.scala:129)
>         at io.circe.syntax.package$EncoderOps$.asJson$extension(package.scala:8)
>         at fpp.compiler.codegen.AnalysisJsonEncoder$.analysisToJson(AnalysisJsonEncoder.scala:355)
>         at fpp.compiler.tools.FPPtoJson$.writeAnalysis$$anonfun$1(fpp-to-json.scala:68)
>         at scala.util.Either.flatMap(Either.scala:352)
>         at fpp.compiler.tools.FPPtoJson$.writeAnalysis(fpp-to-json.scala:69)
>         at fpp.compiler.tools.FPPtoJson$.command$$anonfun$2$$anonfun$1$$anonfun$1$$anonfun$1(fpp-to-json.scala:29)
>         at scala.util.Either.flatMap(Either.scala:352)
>         at fpp.compiler.tools.FPPtoJson$.command$$anonfun$2$$anonfun$1$$anonfun$1(fpp-to-json.scala:30)
>         at scala.util.Either.flatMap(Either.scala:352)
>         at fpp.compiler.tools.FPPtoJson$.command$$anonfun$2$$anonfun$1(fpp-to-json.scala:30)
>         at scala.util.Either.flatMap(Either.scala:352)
>         at fpp.compiler.tools.FPPtoJson$.command$$anonfun$2(fpp-to-json.scala:30)
>         at scala.util.Either.flatMap(Either.scala:352)
>         at fpp.compiler.tools.FPPtoJson$.command(fpp-to-json.scala:30)
>         at fpp.compiler.tools.FPPtoJson$.toolMain$$anonfun$1(fpp-to-json.scala:34)
>         at fpp.compiler.util.Tool.mainMethod(Tool.scala:20)
>         at fpp.compiler.tools.FPPtoJson$.toolMain(fpp-to-json.scala:34)
>         at fpp.compiler.tools.FPP$.command(fpp.scala:24)
>         at fpp.compiler.tools.FPP$.main$$anonfun$1(fpp.scala:37)
>         at fpp.compiler.util.Tool.mainMethod(Tool.scala:20)
>         at fpp.compiler.tools.FPP$.main(fpp.scala:37)
>         at fpp.compiler.tools.FPP.main(fpp.scala)
> Caused by: java.lang.NoSuchFieldException: 0bitmap$1731
>         at java.base@11.0.17/java.lang.Class.getDeclaredField(DynamicHub.java:961)
>         at scala.runtime.LazyVals$.getOffset(LazyVals.scala:103)
>         at fpp.compiler.codegen.AnalysisJsonEncoder$$anon$1871.<clinit>(AnalysisJsonEncoder.scala:299)
>         ... 52 more
> ``` 

 _Originally posted by @bevinduckett in [#4431](https://github.com/nasa/fprime/issues/4431#issuecomment-3548331460)_

--- a/.github/workflows/build-test-rhel8.yml
+++ b/.github/workflows/build-test-rhel8.yml
@@ -38,7 +38,7 @@ jobs:
     - uses: ./.github/actions/setup
     - name: Build Framework
       run: |
-        fprime-util generate
+        fprime-util generate -DFPRIME_ENABLE_JSON_MODEL_GENERATION=ON
         fprime-util build --all -j4
 
   Ref:
@@ -59,7 +59,7 @@ jobs:
     - name: Build Ref
       run: |
         cd Ref
-        fprime-util generate
+        fprime-util generate -DFPRIME_ENABLE_JSON_MODEL_GENERATION=ON
         fprime-util build -j4
 
   UTs:

--- a/.github/workflows/fpp-tests.yml
+++ b/.github/workflows/fpp-tests.yml
@@ -33,7 +33,7 @@ jobs:
       - name: "Generate UT build cache"
         working-directory: ./FppTestProject
         run: |
-          fprime-util generate --ut
+          fprime-util generate --ut -DFPRIME_ENABLE_JSON_MODEL_GENERATION=ON
         shell: bash
       - name: "Build UTs"
         working-directory: ./FppTestProject/FppTest

--- a/.github/workflows/fpp-to-json.yml
+++ b/.github/workflows/fpp-to-json.yml
@@ -30,7 +30,7 @@ jobs:
       - name: "Generate Ref Deployment"
         working-directory: ./Ref
         run: |
-          fprime-util generate
+          fprime-util generate -DFPRIME_ENABLE_JSON_MODEL_GENERATION=ON
         shell: bash
       - name: "Run fpp-to-json on Ref topology"
         working-directory: ./Ref/Top

--- a/cmake/FPrime.cmake
+++ b/cmake/FPrime.cmake
@@ -160,7 +160,12 @@ macro(fprime_initialize_build_system)
 
     # Perform necessary sub-builds
     if (NOT FPRIME_IS_SUB_BUILD)
-        run_sub_build(info-cache target/sub-build/fpp_locs target/sub-build/fpp_depend target/sub-build/fpp_to_json target/sub-build/module_info)
+        set(SUB_BUILD_TARGETS target/sub-build/fpp_locs target/sub-build/fpp_depend)
+        if (FPRIME_ENABLE_JSON_MODEL_GENERATION)
+            list(APPEND SUB_BUILD_TARGETS target/sub-build/fpp_to_json)
+        endif()
+        list(APPEND SUB_BUILD_TARGETS target/sub-build/module_info)
+        run_sub_build(info-cache ${SUB_BUILD_TARGETS})
         # Import the pre-computed properties!
         include("${CMAKE_BINARY_DIR}/fprime_module_info.cmake")
     endif()

--- a/cmake/options.cmake
+++ b/cmake/options.cmake
@@ -148,7 +148,7 @@ option(FPRIME_ENABLE_AUTOCODER_UTS "Enable autocoder UT generation" OFF)
 option(FPRIME_ENABLE_UT_COVERAGE "Calculate unit test coverage" ON)
 
 ####
-# `FPRIME_ENABLE_TEXT_LOGGERS:`
+# `FPRIME_ENABLE_TEXT_LOGGERS`:
 #
 # When FPRIME_ENABLE_TEXT_LOGGERS is set, the ActiveTextLogger and PassiveConsoleTextLogger 
 # svc components are included in the build. When unset, those components are excluded, 
@@ -162,6 +162,20 @@ option(FPRIME_ENABLE_UT_COVERAGE "Calculate unit test coverage" ON)
 # e.g. `-DFPRIME_ENABLE_TEXT_LOGGERS=OFF`
 ####
 option(FPRIME_ENABLE_TEXT_LOGGERS "Enable text loggers in build" ON)
+
+####
+# `FPRIME_ENABLE_JSON_MODEL_GENERATION`:
+#
+# Turns on the generation of JSON models for all modules in the build system. This will run `fpp-to-json` and may
+# require the user to install `java` and the .jar variants of FPP.
+#
+# **Values:**
+# - ON: enable JSON model generation
+# - OFF: (default) disable JSON model generation
+#
+# e.g. `-DFPRIME_ENABLE_JSON_MODEL_GENERATION=ON`
+####
+option(FPRIME_ENABLE_JSON_MODEL_GENERATION "Enable JSON model generation" OFF)
 
 ####
 # `FPRIME_SKIP_TOOLS_VERSION_CHECK`:

